### ISSUE TITLE
Search history

### ISSUE THREAD

**Issue by anthony-bernaert** [DSPaul/COMPASS#85]
Add a dropdown to the search field with the recent text searches to quickly return to previously found files.

Maybe also add general back/forward buttons that act on the include/exclude filters. These would modify the filter one step at a time (in the order they were added). Ideally these buttons would be mapped to the physical back/forward side buttons of a mouse.

**Comment by DSPaul** [comment]
A system similar to Google search that shows your 5 or so most recent searches when you focus the search field that you can go to with arrow keys seems like a good thing to add, I'll add it to the list. 

The going forward & backward for adding filters would be part of a more general undo redo system, already tracked in #30.

**Comment by DSPaul** [comment]
I would argue selecting and removing filters are also actions, rather than a form of navigation because you are staying on the same screen, just modifying its state. So I'm not going to make a separate system for this. It will add bloat and confusion. The undo will simply bring you back to the state before your last click/keyboard shortcut/ ect. which includes undoing adding or removing filters.

**Comment by anthony-bernaert** [comment]
Well, different point of views here :smile:. The reason I mentioned navigation is because I'm intensively using this app as an alternative to the default file explorer. I frequently update files, not only the content on disk, but also the tags, description, ... in-app.

The situation I ran into already a few times is: having certain filters applied, changing the filters to temporarily select another file, editing that file's COMPASS metadata, then going back to the first set of filters.

A unified undo system where data model and view state are combined makes it dangerous to use here, as I'm losing the edit to the metadata when I want to go back.

There may be alternatives to this feature request, perhaps some tabbing system or a derivative. In any case, something to switch filter sets even faster without affecting metadata.

**Comment by DSPaul** [comment]
I think a tab system is indeed the better solution here, as it is a more versatile solution that more people will get use out of.

**Comment by DSPaul** [comment]
Implemented on Avalonia branch, coming in v2.0.0
See commits 341462aa71750509620d0be7324457dddc978429 and d494fc042e0a8c9377eaadcb8980c479faba4950

**Comment by anthony-bernaert** [comment]
Seems like a good compromise indeed. What about removing some rounding of the buttons to emphasize that they are part of a tab system (like the sidebar)?
<img width="1282" height="292" alt="Image" src="https://github.com/user-attachments/assets/3c496cde-002b-4b12-a687-4114c26d5dd0" />

**Comment by DSPaul** [comment]
It's not a bad idea, might help communicate the functionality. However I use firefox as my browser and they just use the same corner radius on all four corners of their tabs so I think both are possible. I'll think about it, might put it to a vote when we get closer to release.

<img width="699" height="105" alt="Image" src="https://github.com/user-attachments/assets/671a7355-a4b4-448c-af2d-d83d4a43bee3" />

**Comment by anthony-bernaert** [comment]
Yeah it's just a detail. Once you know it's there, it'll feel as natural to use. The reason I mentioned this is because I didn't realize at first it was a tab bar -- I think due the presence of the collection edit buttons and the search bar sharing the same space.
